xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1627,27 +1627,26 @@ init(xlator_t *this)
     if (!cmd_history_file) {
         snprintf(cmd_log_filename, PATH_MAX, "%s/cmd_history.log", logdir);
     } else {
-        if (cmd_history_file[0] != '/') {
-            gf_msg(
-                this->name, GF_LOG_ERROR, errno, GD_MSG_FILE_OP_FAILED,
-                "Expects an absolute file for 'command-history-file', got %s",
-                cmd_history_file);
+        if ((cmd_history_file[0] == '/') || (strstr(cmd_history_file, "../"))) {
```

Comment:
This will allow to reference subdirectories below `logdir`. Is that the intention ?

Reply:
@amarts is that your intention, if so, the patch is ok to me.

Reply:
@xhernandez thinking more, and for keeping the goal of 1 PR for 1 task, I would refactor this as 2 PRs. 1 for just handling all logfiles and syslog related. and another one to address this specific thing about `cmd_history_file` as an option etc).

Not in an hurry to get this merged, will rebase and submit again in another 10days. Sorry about delay.